Source/WebKit/WebProcess/Network/WorkerWebTransportReceiveStreamStatsProvider.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WorkerWebTransportReceiveStreamStatsProvider.h"
+
+#include <WebCore/ScriptExecutionContext.h>
+#include <WebCore/WebTransportReceiveStreamStats.h>
+#include <wtf/CompletionHandler.h>
+#include <wtf/RunLoop.h>
+
+namespace WebKit {
+
+Ref<WorkerWebTransportReceiveStreamStatsProvider> WorkerWebTransportReceiveStreamStatsProvider::create(WebCore::ScriptExecutionContextIdentifier identifier, Ref<WebTransportReceiveStreamStatsProvider>&& provider)
+{
+    return adoptRef(*new WorkerWebTransportReceiveStreamStatsProvider(identifier, WTFMove(provider)));
+}
+
+WorkerWebTransportReceiveStreamStatsProvider::WorkerWebTransportReceiveStreamStatsProvider(WebCore::ScriptExecutionContextIdentifier identifier, Ref<WebTransportReceiveStreamStatsProvider>&& provider)
+    : m_scriptExecutionContextIdentifier(identifier)
+    , m_provider(WTFMove(provider))
+{
+    ASSERT(!RunLoop::isMain());
+}
+
+WorkerWebTransportReceiveStreamStatsProvider::~WorkerWebTransportReceiveStreamStatsProvider()
+{
+    ASSERT(!RunLoop::isMain());
+}
+
+void WorkerWebTransportReceiveStreamStatsProvider::getStats(CompletionHandler<void(std::optional<WebCore::WebTransportReceiveStreamStats>&&)>&& completionHandler)
+{
+    ASSERT(!RunLoop::isMain());
+    RunLoop::main().dispatch([provider = m_provider, identifier = m_scriptExecutionContextIdentifier, completionHandler = WTFMove(completionHandler)] () mutable {
```

Comment:
Given we might want to do networking from a worker thread without going to main thread, I wonder whether getStats will not follow the same path.
How about removing this code for now?